File: finops-adx/clients/python/query_costs.py
from azure.kusto.data import KustoClient, KustoConnectionStringBuilder

cluster = "https://finops-nar-ade.centralus.kusto.windows.net"
database = "Hub"

kcsb = KustoConnectionStringBuilder.with_az_cli_authentication(cluster)
client = KustoClient(kcsb)

query = "Costs | take 5"
response = client.execute(database, query)

for row in response.primary_results[0]:
    print(row)

File: finops-adx/clients/typescript/query-costs.ts
import { KustoConnectionStringBuilder, Client } from "azure-kusto-data";

const cluster = "https://finops-nar-ade.centralus.kusto.windows.net";
const database = "Hub";

const kcsb = KustoConnectionStringBuilder.withAadDeviceAuthentication(cluster); 
const client = new Client(kcsb);

async function main() {
    const query = "Costs | take 5";
    const response = await client.execute(database, query);
    console.log(response.primaryResults[0]?.toString());
}

main();